ggshield/cmd/config/config_list.py

```diff
@@ -54,16 +56,34 @@ def add_entries(*entries: Tuple[str, Any]):
             else default_token_lifetime
         )
 
-        message_lines.append(f"[{instance_name}]")
-        add_entries(
-            ("default_token_lifetime", _default_token_lifetime),
-            ("workspace_id", workspace_id),
-            ("url", instance.url),
-            ("token", token),
-            ("token_name", token_name),
-            ("expiry", expiry),
+        instance_info = {
+            "default_token_lifetime": _default_token_lifetime,
+            "workspace_id": workspace_id,
+            "url": instance.url,
+            "token": token,
+            "token_name": token_name,
+            "expiry": expiry,
+        }
+        config_data["instances"].append(
+            {"instance_name": instance_name, **instance_info}
         )
+
+    if ctx_obj.use_json:
+        click.echo(json.dumps(config_data, indent=2))
```

Comment:
No need for `indent=2`: this is for machines, and non machines can pipe it to `jq` :)